compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -228,26 +228,36 @@ static int32_t getLongestPathOfDAG(TR::Node *node, LongestPathMap &memo)
    return value;
    }
 
-static bool containsNode(TR::Node *containerNode, TR::Node *node, vcount_t visitCount, TR::Compilation *comp, int32_t *height, int32_t *maxHeight, bool &canMoveIfVolatile)
+/*
+ * \brief This function does 2 things:
+ *        1. check if the \p nodeToSwingDown is in the subtree of \p containerNode
+ *        2. only allowing volatile load being swung down across store/load of temps
+ *
+ * \parm nodeToSwingDown
+ *    The node hanging under a treetop which can potentially be removed
+ *
+ * \parm containerNode
+ *    The subtree to check
+ */
+static bool containsNode(TR::Node *containerNode, TR::Node *nodeToSwingDown, vcount_t visitCount, TR::Compilation *comp, int32_t *height, int32_t *maxHeight, bool &canMoveIfVolatile)
    {
-   if (containerNode == node)
+   if (containerNode == nodeToSwingDown)
       return true;
 
    vcount_t oldVisitCount = containerNode->getVisitCount();
    if ((oldVisitCount == visitCount) || (oldVisitCount == comp->getVisitCount()))
       return false;
    containerNode->setVisitCount(comp->getVisitCount());
 
-   if (containerNode->getOpCode().hasSymbolReference() &&
-       (containerNode->getSymbol()->isShadow() || containerNode->getSymbol()->isStatic()))
+   if (containerNode->getOpCode().hasSymbolReference() && !containerNode->getSymbol()->isAuto())
```

Comment:
Can you explain why this is so much more strict? There are probably non-helper symbols etc that it is safe to swing across - what are the cases we don't want to swing across? Methods? something else? This seems quite draconian so I'm just wanting to understand why you chose such a strict check.

Reply:
@vijaysun-omr mentioned when he added the optimization he was mainly thinking of swinging down across temps. For other symbols, I think we can't apply the opt for call, check, shadow. I feel swinging down volatile is generally risky so it's safer to change the condition to only include the case we are sure of.

Reply:
Right I guess the risk is a perf regression - if we are going to make it more strict could we leave an envVar to test the old behaviour (at least for now) with a comment that the version of the check it enables is incorrect. That way we can at least rule this change in/out of regressions easily?

Reply:
added env var to guard the change.